apps/web/src/pages/templates/components/SmsPreview.tsx

```diff
@@ -0,0 +1,95 @@
+import styled from '@emotion/styled';
+import { colors } from '@novu/design-system';
+import { useAuthController, useDataRef } from '@novu/shared-web';
+import { useEffect, useState } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+
+import { useGetLocalesFromContent, usePreviewSms } from '../../../api/hooks';
+import { LocaleSelect } from '../../../components/workflow/Preview/common';
+import { useNavigateToStepEditor } from '../hooks/useNavigateToStepEditor';
+import { useStepFormErrors } from '../hooks/useStepFormErrors';
+import { useStepFormPath } from '../hooks/useStepFormPath';
+import { IForm } from './formTypes';
+import { MobileSimulator } from './phone-simulator';
+import { SmsBubble } from './SmsBubble';
+
+const BodyContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  margin: auto 20px 40px 20px;
+`;
+
+const LocaleSelectStyled = styled(LocaleSelect)`
+  .mantine-Select-input {
+    color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B60)};
+  }
+
+  .mantine-Input-rightSection {
+    color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B60)} !important;
+  }
+`;
+
+export const SmsPreview = () => {
+  const { control } = useFormContext<IForm>();
+  const path = useStepFormPath();
+  const error = useStepFormErrors();
+  const templateContent = useWatch({
+    name: `${path}.template.content`,
+    control,
+  });
+  const { organization } = useAuthController();
+  const [previewContent, setPreviewContent] = useState(templateContent as string);
+  const [selectedLocale, setSelectedLocale] = useState('');
```

Comment:
`previewContent` is a compiled template content shown as the text
`selectedLocale` is the locale to which the template would be compiled

Reply:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** These would be helpful in code too!‏